components/core/src/package/archive.rs

```diff
@@ -417,6 +419,95 @@ pub trait FromArchive: Sized {
+
+    fn try_from(mut archive: PackageArchive) -> Result<Self> {
+        let header = artifact::get_artifact_header(&archive.path)?;
+        let ident = archive.ident()?;
+        if ident.fully_qualified() {
+            Ok(PackageArchiveInfo { format_version: header.format_version,
+                                    key_name:       header.key_name,
+                                    hash_type:      header.hash_type,
+                                    signature_raw:  header.signature_raw,
+                                    origin:         ident.origin.clone(),
+                                    name:           ident.name.clone(),
+                                    version:        ident.version.unwrap_or_default(),
+                                    release:        ident.release.unwrap_or_default(),
+                                    checksum:       archive.checksum()?,
+                                    target:         format!("{}", archive.target()?),
+                                    is_a_service:   archive.is_a_service(),
+                                    deps:           archive.deps()
+                                                           .unwrap_or_default()
+                                                           .iter()
+                                                           .map(|x| format!("{}", x))
+                                                           .collect(),
+                                    build_deps:     archive.build_deps()
+                                                           .unwrap_or_default()
+                                                           .iter()
+                                                           .map(|x| format!("{}", x))
+                                                           .collect(),
+                                    tdeps:          archive.tdeps()
+                                                           .unwrap_or_default()
+                                                           .iter()
+                                                           .map(|x| format!("{}", x))
+                                                           .collect(),
+                                    build_tdeps:    archive.build_tdeps()
+                                                           .unwrap_or_default()
+                                                           .iter()
+                                                           .map(|x| format!("{}", x))
+                                                           .collect(),
+                                    exposes:        archive.exposes().unwrap_or_default(),
+                                    manifest:       archive.manifest()?.to_string(),
+                                    svc_user:       archive.svc_user()
+                                                           .ok()
+                                                           .map(ToString::to_string),
```

Comment:
Kind of interesting that `archive.svc_user()` returns a `Result` rather than an `Option<Result>`, seeing as how packages can legitimately not have service users :thinking: 

(Just an observation; not necessarily anything actionable right now :sweat_smile:)

Reply:
That was strange indeed. I have changed `archive.svc_user()` to return an `Option<&str>` instead of `Result<&str>` and added `archive.svc_group()` which does the same. https://github.com/habitat-sh/habitat/pull/7814/commits/5e1bee29f1f984a482a0cccef3b1819e8fc1e240